int32 Width
int32 Height